dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/DhisConvenienceTest.java

```diff
@@ -2285,6 +2285,29 @@ protected User createUser( String username, String... authorities )
         return user;
     }
 
+    protected User createUserWithOpenID( String username, String openIDIdentifier, String... authorities )
+    {
+        checkUserServiceWasInjected();
+
+        UserAuthorityGroup group = createAuthorityGroup( username, authorities );
+
+        userService.addUserAuthorityGroup( group );
+
+        User user = createUser( username );
+
+        userService.addUser( user );
+
+        UserCredentials credentials = createUserCredentialsWithOpenID( openIDIdentifier, username, user, group );
+
+        userService.encodeAndSetPassword( credentials, DEFAULT_ADMIN_PASSWORD );
+        userService.addUserCredentials( credentials );
+
+        user.setUserCredentials( credentials );
+        userService.updateUser( user );
+
+        return user;
```

Comment:
Pretty sure there are existing methods in the class that already do most of what this is doing. There should be something you can reuse and then this method does the rest that is special to openID login (maybe it comes down to `createUserCredentialsWithOpenID`?)

Reply:
Agree, I will refactor it. It's only one line different.